WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.kt

```diff
@@ -3526,6 +3529,20 @@ class EditPostActivity : LocaleAwareActivity(), EditorFragmentActivity, EditorIm
         // Start VM, load prompt and populate Editor with content after edit IS ready.
         val promptId: Int = intent.getIntExtra(EditPostActivityConstants.EXTRA_PROMPT_ID, -1)
         editorBloggingPromptsViewModel.start(siteModel, promptId)
+
+        updateVoiceContentIfNeeded()
+    }
+
+    private fun updateVoiceContentIfNeeded() {
+        // Check if voice content exists and this is a new post for a Gutenberg editor fragment
+        val content = intent.getStringExtra(EditPostActivityConstants.EXTRA_VOICE_CONTENT)
+        if (isNewPost && content != null && !isVoiceContentSet) {
```

Comment:
Do we need to track this `isVoiceContentSet` with a boolean? Is there a case this `updateVoiceContentIfNeeded()` is being called multiple times for a single `EditPostActivity` instance?

Reply:
@irfano To prevent the content from reloading before the post is saved - could happen on return from background.